src/main/java/seedu/jarvis/logic/parser/planner/AddTaskCommandParser.java

```diff
@@ -53,11 +53,15 @@ public AddTaskCommand parse(String args) throws ParseException {
         String taskType = argMultimap.getValue(PREFIX_TASK_TYPE).get();
         String taskDes = argMultimap.getValue(PREFIX_TASK_DES).get();
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
-        System.out.println("hello there");
         LocalDate[] dates = new LocalDate[2];
         if (isEventOrDeadline(argMultimap)) {
-            System.out.println("hi there");
             dates = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
+            if (taskType.equals("event")) {
+                if (dates[1] == null) {
```

Comment:
Your way of parsing dates in `ParserUtil` is quite manual and could be abused. You can look up formatters to apply to the date to parse strings. They are quite handy in handling parsing date strings with a degree of control over how you "format" the date